### ISSUE TITLE
Time Out Exceptions

### ISSUE THREAD

**Issue by harrigan** [cassandra-rb/cassandra#25]
Hi,

I'm not sure if this is a problem with the Cassandra gem, the Thrift gem, Thrift, or Cassanda, but I'll start at the top and work down ;-)

My script is as follows:

```
require 'rubygems'
require 'cassandra'
require 'simple_uuid'
include SimpleUUID

client = Cassandra.new 'MyKeyspace', '127.0.0.1:9160'
client.clear_keyspace!

def random_string length = 20
  chars = 'abcdefghjkmnpqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ23456789'
  string = ''
  length.times do
    string << chars[rand(chars.size)]
  end
  string
end

count = 0
last_uuid = nil
while true
  count += 1
  print '.' if count % 1000 == 0
  client.get : MyData, last_uuid unless last_uuid.nil?
  current_uuid = UUID.new.to_guid
  client.insert : MyData, current_uuid, {
    'data_0' => random_string,
    # snip
    'data_9' => random_string
  }
  last_uuid = current_uuid
end
```

After about a minute:
/lib/ruby/gems/1.9.1/gems/thrift-0.2.0.2/lib/thrift/transport/socket.rb:108:in `read': Socket: Timed out reading 4096 bytes from 127.0.0.1:9160 (Thrift::TransportException)

The exception is not raised when I omit the data columns.

I am using Cassandra 0.6.1 and the Cassandra gem 0.8.2.

Any help much appreciated.

**Comment by ryanking** [comment]
where in your script is the exception being raised?

**Comment by harrigan** [comment]
At the client.insert : MyData... line.

If I rescue the exception, sleep for some seconds and retry the insert, then it works. Can a single script like this overload a Cassandra node?

**Comment by jmhodges** [comment]
It could be if the amount of memory you gave the JVM and the Memtable thresholds you set are off. http://wiki.apache.org/cassandra/FAQ#slows_down_after_lotso_inserts

**Comment by wouterdebie** [comment]
I'm experiencing exactly the same thing. Running the latest gem and Cassandra 0.6.1 on Ruby 1.8.7. When it happens is completely random. Sometimes after just a couple inserts, sometimes after a 100000. I'm doing almost the same thing as the script above. Any help would be appreciated.

**Comment by harrigan** [comment]
I increased the JVM's heap from 1GB to 2GB. Now the exception occurs less frequently but it still occurs. I'll try changing the Memtable thresholds also. I'm just surprised that the script above overloads a single node so quickly.

**Comment by ldipenti** [comment]
Hello all, I experienced this also and asked on the Cassandra mailing list. My solution was:

client = Cassandra.new 'MyKeyspace', '127.0.0.1:9160', :timeout => 10.0

Where :timeout is in seconds.

My test setup is: Cassandra 0.6.1, Cassandra gem 0.8.2 on a macbook w/4GB RAM and all configurations on default.

When I did the tests, it seems that a Timeout exception is raised when compaction takes too long.
